app/models/miq_worker/deployment_per_worker.rb

```diff
@@ -5,7 +5,6 @@ module DeploymentPerWorker
     def create_container_objects
       ContainerOrchestrator.new.create_deployment(worker_deployment_name) do |definition|
         configure_worker_deployment(definition, 1)
-        definition[:spec][:template][:spec][:containers].first[:env] << {:name => "EMS_ID", :value => self.class.ems_id_from_queue_name(queue_name)}
```

Comment:
This is what bothered me originally, we already have EMS_ID env var handling for systemd in the per_ems_worker_mixin so I thought we should be able to let the worker class define this rather than `DeploymentPerWorker`